packages/bpk-component-floating-notification/src/BpkFloatingNotification.tsx

```diff
@@ -17,58 +17,60 @@
  */
 /* @flow strict */
 
-import PropTypes from 'prop-types';
-import { useEffect, useState, ReactElement } from 'react';
-import { CSSTransition } from 'react-transition-group';
+import type { MouseEvent, FunctionComponent } from 'react';
+import { useEffect, useState } from 'react';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
+import CSSTransition from 'react-transition-group/CSSTransition';
 
 import BpkAriaLive from '../../bpk-component-aria-live';
-import {BUTTON_TYPES, BpkButtonV2} from '../../bpk-component-button';
+import { BUTTON_TYPES, BpkButtonV2 } from '../../bpk-component-button';
 import BpkText, { TEXT_STYLES } from '../../bpk-component-text';
 import { cssModules } from '../../bpk-react-utils';
 
 import STYLES from './BpkFloatingNotification.module.scss';
 
 const getClassName = cssModules(STYLES);
 
-type Props = {
-  animateOnEnter: ?boolean,
-  animateOnExit: ?boolean,
-  className: ?string,
-  ctaText: ?string,
+export type Props = {
+  animateOnEnter?: boolean;
+  animateOnExit?: boolean;
+  className?: string;
+  ctaText?: string;
   /**
    * This prop controls the amount of time that the notification stays visible before the exit animation begins.
    * The default value is 4 seconds (4000 milliseconds).
    */
-  hideAfter: ?number,
-  icon: ?() => ReactElement,
-  onClick: ?() => void,
+  hideAfter?: number;
+  icon?: FunctionComponent;
+  onClick?: (e: MouseEvent<HTMLButtonElement>) => void;
   /**
    * Execute a function after the component has finished the exit animation.
    */
-  onExit: ?() => void,
-  text: string,
+  onExit?: () => void;
+  text: string;
+  [rest: string]: any; // Inexact rest. See decisions/inexact-rest.md
 };
 
 const BpkFloatingNotification = (props: Props) => {
   const [showMessage, setShowMessage] = useState(true);
 
   const {
-    animateOnEnter,
-    animateOnExit,
-    className,
-    ctaText,
-    hideAfter,
+    animateOnEnter = true,
+    animateOnExit = true,
+    className = null,
+    ctaText = null,
+    hideAfter = 4000,
     icon: Icon,
     onClick,
-    onExit,
+    onExit = null,
```

Comment:
Nit: could these null values be left undefined? Actually I'm surprised it doesn't cause type issues since the type definition doesn't have null has a type.

Reply:
Sure, they are completely unnecessary. I'll remove them.